# Gene Set Enrichment Module

This functionality is an extension to the differential gene identification. Users are able to use different cutoffs to filter out DEGs/DEPs/DEMs in different comparisons and probe enriched pathways. This is a useful way to make biological sense of the data. Two types of the enrichment analysis are available: gene set enrichment analysis and over-representation analysis. 

## Gene Set Enrichment Analysis (GSEA)

This first tab lets the users pick a particular comparison, select a species pathway database, set analysis statistical cutoff criteria and select gene set database(s) from a list of different databases for gene set enrichment analysis(GSEA). 

[![](`r pic_path`f9-1.png)](`r pic_path`f9-1.png)

## Over-Representation Analysis (ORA)

This tab lets the users to test a gene list by either picking a particular comparison with the comparison cutoffs set or providing a customized gene list. Then select a species pathway database, set analysis statistical cutoff criteria and select select gene set database(s) from a list of different databases for over-representation analysis(ORA). 

[![](`r pic_path`f9-2.png)](`r pic_path`f9-2.png)

## Gene Expression

For the enriched terms identified in the previous tabs (GSEA and ORA), users have the ability to look at the differential values of the genes present in the set. Here we are probing the Phagosome gene set identified as the top 1 enriched gene set in the GSEA analysis.

[![](`r pic_path`f9-3.png)](`r pic_path`f9-3.png)

The page looks a bit different for the ORA analysis result. It does not have the enrichment plot.

[![](`r pic_path`f9-4.png)](`r pic_path`f9-4.png)

## Gene Set Heatmap

Like the “Gene Expression” tab, users have the ability to plot expression of the genes in the pathway term selected as a heatmap. The plot can use all the genes in the gene set or just the leading/DE genes.This visualization shows that almost all the Phagosome genes are enriched in the same direction. 

[![](`r pic_path`f9-5.png)](`r pic_path`f9-5.png)


## KEGG Pathway View

This functionality is available only to KEGG pathway terms. Users have the option to view the Fold Change levels of different genes in the KEGG pathway map. If only one type of source comparison is used, users can select for up to 5 comparisons. 

[![](`r pic_path`f9-6.png)](`r pic_path`f9-6.png)

Or users can select 2 comparison from different data sources (RNAseq, proteomics or metabolomics data). In this example we demo the KEGG pathway “mmu05230 central carbon metabolism in cancer” as identified as top 1 gene set in the Metabolite_HFvsCD_Old comparison. 

[![](`r pic_path`f9-7.png)](`r pic_path`f9-7.png)


## WikiPathway View

This functionality is similar to the "KEGG Pathway View" but it is for the gene set visualization in the WikiPathways database. In this example we demo the Wiki pathway "electron transport chain WP395" as identified as top 1 gene set in the RNAseq_OldvsYoung_HF comparison. It clearly shows that almost all the genes in the gene set are down-regulated.

[![](`r pic_path`f9-8.png)](`r pic_path`f9-8.png)
