<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 1 Introduction | xOmicsShiny: an R shiny application for cross-omics data analysis and pathway mapping</title>
  <meta name="description" content="Chapter 1 Introduction | xOmicsShiny: an R shiny application for cross-omics data analysis and pathway mapping" />
  <meta name="generator" content="bookdown 0.36 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 1 Introduction | xOmicsShiny: an R shiny application for cross-omics data analysis and pathway mapping" />
  <meta property="og:type" content="book" />
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 1 Introduction | xOmicsShiny: an R shiny application for cross-omics data analysis and pathway mapping" />
  
  
  



<meta name="date" content="2024-07-22" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="index.html"/>
<link rel="next" href="installation.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>



<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">xOmicsShiny</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Preface</a></li>
<li class="chapter" data-level="1" data-path="introduction.html"><a href="introduction.html"><i class="fa fa-check"></i><b>1</b> Introduction</a></li>
<li class="chapter" data-level="2" data-path="installation.html"><a href="installation.html"><i class="fa fa-check"></i><b>2</b> Installation</a></li>
<li class="chapter" data-level="3" data-path="dataset-module.html"><a href="dataset-module.html"><i class="fa fa-check"></i><b>3</b> Dataset Module</a>
<ul>
<li class="chapter" data-level="3.1" data-path="dataset-module.html"><a href="dataset-module.html#select-datasets"><i class="fa fa-check"></i><b>3.1</b> Select Datasets</a>
<ul>
<li class="chapter" data-level="3.1.1" data-path="dataset-module.html"><a href="dataset-module.html#saved-projects-in-csv-file"><i class="fa fa-check"></i><b>3.1.1</b> Saved Projects in CSV file</a></li>
<li class="chapter" data-level="3.1.2" data-path="dataset-module.html"><a href="dataset-module.html#upload-data-files-csv-and-upload-rdata-file"><i class="fa fa-check"></i><b>3.1.2</b> Upload Data Files (csv) and Upload RData File</a></li>
<li class="chapter" data-level="3.1.3" data-path="dataset-module.html"><a href="dataset-module.html#load-and-unload-datasets"><i class="fa fa-check"></i><b>3.1.3</b> Load and Unload datasets</a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="dataset-module.html"><a href="dataset-module.html#sample-table"><i class="fa fa-check"></i><b>3.2</b> Sample Table</a></li>
<li class="chapter" data-level="3.3" data-path="dataset-module.html"><a href="dataset-module.html#result-table"><i class="fa fa-check"></i><b>3.3</b> Result Table</a></li>
<li class="chapter" data-level="3.4" data-path="dataset-module.html"><a href="dataset-module.html#data-table"><i class="fa fa-check"></i><b>3.4</b> Data Table</a></li>
<li class="chapter" data-level="3.5" data-path="dataset-module.html"><a href="dataset-module.html#protein-gene-names"><i class="fa fa-check"></i><b>3.5</b> Protein Gene Names</a></li>
<li class="chapter" data-level="3.6" data-path="dataset-module.html"><a href="dataset-module.html#help"><i class="fa fa-check"></i><b>3.6</b> Help</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="groups-and-samples-module.html"><a href="groups-and-samples-module.html"><i class="fa fa-check"></i><b>4</b> Groups and Samples Module</a>
<ul>
<li class="chapter" data-level="4.1" data-path="groups-and-samples-module.html"><a href="groups-and-samples-module.html#select-gourps-and-samples"><i class="fa fa-check"></i><b>4.1</b> Select gourps and samples</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="qc-plots-module.html"><a href="qc-plots-module.html"><i class="fa fa-check"></i><b>5</b> QC Plots Module</a>
<ul>
<li class="chapter" data-level="5.1" data-path="qc-plots-module.html"><a href="qc-plots-module.html#pca-plot"><i class="fa fa-check"></i><b>5.1</b> PCA Plot</a></li>
<li class="chapter" data-level="5.2" data-path="qc-plots-module.html"><a href="qc-plots-module.html#eigenvalues"><i class="fa fa-check"></i><b>5.2</b> Eigenvalues</a></li>
<li class="chapter" data-level="5.3" data-path="qc-plots-module.html"><a href="qc-plots-module.html#pca-loading-table"><i class="fa fa-check"></i><b>5.3</b> PCA Loading Table</a></li>
<li class="chapter" data-level="5.4" data-path="qc-plots-module.html"><a href="qc-plots-module.html#covariates"><i class="fa fa-check"></i><b>5.4</b> Covariates</a></li>
<li class="chapter" data-level="5.5" data-path="qc-plots-module.html"><a href="qc-plots-module.html#alignqc"><i class="fa fa-check"></i><b>5.5</b> AlignQC</a></li>
<li class="chapter" data-level="5.6" data-path="qc-plots-module.html"><a href="qc-plots-module.html#pca-3d-plot"><i class="fa fa-check"></i><b>5.6</b> PCA 3D Plot</a></li>
<li class="chapter" data-level="5.7" data-path="qc-plots-module.html"><a href="qc-plots-module.html#pca-3d-interactive"><i class="fa fa-check"></i><b>5.7</b> PCA 3D Interactive</a></li>
<li class="chapter" data-level="5.8" data-path="qc-plots-module.html"><a href="qc-plots-module.html#sample-sample-distance"><i class="fa fa-check"></i><b>5.8</b> Sample-sample Distance</a></li>
<li class="chapter" data-level="5.9" data-path="qc-plots-module.html"><a href="qc-plots-module.html#dendrograms"><i class="fa fa-check"></i><b>5.9</b> Dendrograms</a></li>
<li class="chapter" data-level="5.10" data-path="qc-plots-module.html"><a href="qc-plots-module.html#box-plot"><i class="fa fa-check"></i><b>5.10</b> Box Plot</a></li>
<li class="chapter" data-level="5.11" data-path="qc-plots-module.html"><a href="qc-plots-module.html#cv-distribution"><i class="fa fa-check"></i><b>5.11</b> CV Distribution</a></li>
<li class="chapter" data-level="5.12" data-path="qc-plots-module.html"><a href="qc-plots-module.html#help-1"><i class="fa fa-check"></i><b>5.12</b> Help</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">xOmicsShiny: an R shiny application for cross-omics data analysis and pathway mapping</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="introduction" class="section level1 hasAnchor" number="1">
<h1><span class="header-section-number">Chapter 1</span> Introduction<a href="introduction.html#introduction" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>xOmicsShiny can be accessed via the link <a href="https://xomicsshiny.bxgenomics.com/" target="_blank">https://xomicsshiny.bxgenomics.com/</a>. It is an R shiny application for cross-omics data analysis and pathway mapping. RNAseq, Proteomics and metabolomics data are supported in both individual and integrated way to visualize statistical analysis results. This tutorial provides a detailed guide on using the different functionalities and customizing the tools to best fit data analysis and plotting needs, using published multiomic dataset as examples (<a href="https://doi.org/10.1016/j.cels.2021.09.005" target="_blank">Williams et al., 2022</a>).</p>
<p><a href="Figures/f1-1.png"><img src="Figures/f1-1.png" /></a></p>
<p>The main interface consists of two tabs: <strong>Dataset</strong> and <strong>Setting</strong>. The <strong>Dataset</strong> page features a multi-tab menu for various functionality modules. Before loading a dataset, the default tabs include Select Dataset, four tabs displaying dataset information, and a Help tab explaining the sub-tab functions.</p>
<p><a href="Figures/f1-2.png"><img src="Figures/f1-2.png" /></a></p>
<p>The <strong>Setting</strong> page lists all available functional modules. Users can select or deselect individual modules by clicking on the module icon, which displays a text prompt on hover. Selected module names will be added as tabs in the multi-tab menu on the <strong>Dataset</strong> page.</p>
<p><a href="Figures/f1-3.png"><img src="Figures/f1-3.png" /></a></p>

</div>
            </section>

          </div>
        </div>
      </div>
<a href="index.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="installation.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/interactivereport/xOmicsShiny/edit/gh-pages/01-Introduction.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": "https://github.com/interactivereport/xOmicsShiny/blob/gh-pages/01-Introduction.Rmd",
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
