<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 11 WGCNA Module | xOmicsShiny: an R shiny application for cross-omics data analysis and pathway mapping</title>
  <meta name="description" content="Chapter 11 WGCNA Module | xOmicsShiny: an R shiny application for cross-omics data analysis and pathway mapping" />
  <meta name="generator" content="bookdown 0.36 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 11 WGCNA Module | xOmicsShiny: an R shiny application for cross-omics data analysis and pathway mapping" />
  <meta property="og:type" content="book" />
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 11 WGCNA Module | xOmicsShiny: an R shiny application for cross-omics data analysis and pathway mapping" />
  
  
  



<meta name="date" content="2025-01-31" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="pattern-clustering-module.html"/>
<link rel="next" href="correlation-network-module.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">xOmicsShiny</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Preface</a></li>
<li class="chapter" data-level="1" data-path="introduction.html"><a href="introduction.html"><i class="fa fa-check"></i><b>1</b> Introduction</a></li>
<li class="chapter" data-level="2" data-path="installation.html"><a href="installation.html"><i class="fa fa-check"></i><b>2</b> Installation</a>
<ul>
<li class="chapter" data-level="2.1" data-path="installation.html"><a href="installation.html#set-up-r-environment"><i class="fa fa-check"></i><b>2.1</b> Set up R environment</a></li>
<li class="chapter" data-level="2.2" data-path="installation.html"><a href="installation.html#set-up-configure-files"><i class="fa fa-check"></i><b>2.2</b> Set up configure files</a></li>
<li class="chapter" data-level="2.3" data-path="installation.html"><a href="installation.html#set-up-gene-sets"><i class="fa fa-check"></i><b>2.3</b> Set up gene sets</a></li>
<li class="chapter" data-level="2.4" data-path="installation.html"><a href="installation.html#metabase-users"><i class="fa fa-check"></i><b>2.4</b> Metabase Users</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="dataset-module.html"><a href="dataset-module.html"><i class="fa fa-check"></i><b>3</b> Dataset Module</a>
<ul>
<li class="chapter" data-level="3.1" data-path="dataset-module.html"><a href="dataset-module.html#select-datasets"><i class="fa fa-check"></i><b>3.1</b> Select Datasets</a>
<ul>
<li class="chapter" data-level="3.1.1" data-path="dataset-module.html"><a href="dataset-module.html#saved-projects-in-csv-file"><i class="fa fa-check"></i><b>3.1.1</b> Saved Projects in CSV file</a></li>
<li class="chapter" data-level="3.1.2" data-path="dataset-module.html"><a href="dataset-module.html#upload-data-files-csv-and-upload-rdata-file"><i class="fa fa-check"></i><b>3.1.2</b> Upload Data Files (csv) and Upload RData File</a></li>
<li class="chapter" data-level="3.1.3" data-path="dataset-module.html"><a href="dataset-module.html#load-and-unload-datasets"><i class="fa fa-check"></i><b>3.1.3</b> Load and Unload datasets</a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="dataset-module.html"><a href="dataset-module.html#sample-table"><i class="fa fa-check"></i><b>3.2</b> Sample Table</a></li>
<li class="chapter" data-level="3.3" data-path="dataset-module.html"><a href="dataset-module.html#result-table"><i class="fa fa-check"></i><b>3.3</b> Result Table</a></li>
<li class="chapter" data-level="3.4" data-path="dataset-module.html"><a href="dataset-module.html#data-table"><i class="fa fa-check"></i><b>3.4</b> Data Table</a></li>
<li class="chapter" data-level="3.5" data-path="dataset-module.html"><a href="dataset-module.html#protein-gene-names"><i class="fa fa-check"></i><b>3.5</b> Protein Gene Names</a></li>
<li class="chapter" data-level="3.6" data-path="dataset-module.html"><a href="dataset-module.html#help"><i class="fa fa-check"></i><b>3.6</b> Help</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="groups-and-samples-module.html"><a href="groups-and-samples-module.html"><i class="fa fa-check"></i><b>4</b> Groups and Samples Module</a>
<ul>
<li class="chapter" data-level="4.1" data-path="groups-and-samples-module.html"><a href="groups-and-samples-module.html#select-gourps-and-samples"><i class="fa fa-check"></i><b>4.1</b> Select gourps and samples</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="qc-plots-module.html"><a href="qc-plots-module.html"><i class="fa fa-check"></i><b>5</b> QC Plots Module</a>
<ul>
<li class="chapter" data-level="5.1" data-path="qc-plots-module.html"><a href="qc-plots-module.html#pca-plot"><i class="fa fa-check"></i><b>5.1</b> PCA Plot</a>
<ul>
<li class="chapter" data-level="5.1.1" data-path="qc-plots-module.html"><a href="qc-plots-module.html#pca-plot-1"><i class="fa fa-check"></i><b>5.1.1</b> PCA Plot</a></li>
<li class="chapter" data-level="5.1.2" data-path="qc-plots-module.html"><a href="qc-plots-module.html#eigenvalues"><i class="fa fa-check"></i><b>5.1.2</b> Eigenvalues</a></li>
<li class="chapter" data-level="5.1.3" data-path="qc-plots-module.html"><a href="qc-plots-module.html#pca-loading-table"><i class="fa fa-check"></i><b>5.1.3</b> PCA Loading Table</a></li>
<li class="chapter" data-level="5.1.4" data-path="qc-plots-module.html"><a href="qc-plots-module.html#pca-3d-plot"><i class="fa fa-check"></i><b>5.1.4</b> PCA 3D Plot</a></li>
<li class="chapter" data-level="5.1.5" data-path="qc-plots-module.html"><a href="qc-plots-module.html#pca-3d-interactive"><i class="fa fa-check"></i><b>5.1.5</b> PCA 3D Interactive</a></li>
</ul></li>
<li class="chapter" data-level="5.2" data-path="qc-plots-module.html"><a href="qc-plots-module.html#covariates"><i class="fa fa-check"></i><b>5.2</b> Covariates</a></li>
<li class="chapter" data-level="5.3" data-path="qc-plots-module.html"><a href="qc-plots-module.html#alignqc"><i class="fa fa-check"></i><b>5.3</b> AlignQC</a></li>
<li class="chapter" data-level="5.4" data-path="qc-plots-module.html"><a href="qc-plots-module.html#sample-sample-distance"><i class="fa fa-check"></i><b>5.4</b> Sample-sample Distance</a></li>
<li class="chapter" data-level="5.5" data-path="qc-plots-module.html"><a href="qc-plots-module.html#dendrograms"><i class="fa fa-check"></i><b>5.5</b> Dendrograms</a></li>
<li class="chapter" data-level="5.6" data-path="qc-plots-module.html"><a href="qc-plots-module.html#box-plot"><i class="fa fa-check"></i><b>5.6</b> Box Plot</a></li>
<li class="chapter" data-level="5.7" data-path="qc-plots-module.html"><a href="qc-plots-module.html#cv-distribution"><i class="fa fa-check"></i><b>5.7</b> CV Distribution</a></li>
<li class="chapter" data-level="5.8" data-path="qc-plots-module.html"><a href="qc-plots-module.html#help-1"><i class="fa fa-check"></i><b>5.8</b> Help</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="deg-module.html"><a href="deg-module.html"><i class="fa fa-check"></i><b>6</b> DEG Module</a>
<ul>
<li class="chapter" data-level="6.1" data-path="deg-module.html"><a href="deg-module.html#deg-counts"><i class="fa fa-check"></i><b>6.1</b> DEG Counts</a></li>
<li class="chapter" data-level="6.2" data-path="deg-module.html"><a href="deg-module.html#volcano-plot-static"><i class="fa fa-check"></i><b>6.2</b> Volcano Plot (Static)</a></li>
<li class="chapter" data-level="6.3" data-path="deg-module.html"><a href="deg-module.html#volcano-plot-interactive"><i class="fa fa-check"></i><b>6.3</b> Volcano Plot (Interactive)</a></li>
<li class="chapter" data-level="6.4" data-path="deg-module.html"><a href="deg-module.html#degs-and-deps-in-two-comparisons"><i class="fa fa-check"></i><b>6.4</b> DEGs and DEPs in Two Comparisons</a></li>
<li class="chapter" data-level="6.5" data-path="deg-module.html"><a href="deg-module.html#mulitple-volcano-plots"><i class="fa fa-check"></i><b>6.5</b> Mulitple Volcano Plots</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="heatmap-module.html"><a href="heatmap-module.html"><i class="fa fa-check"></i><b>7</b> Heatmap Module</a>
<ul>
<li class="chapter" data-level="7.1" data-path="heatmap-module.html"><a href="heatmap-module.html#static-heatmap-layout-1"><i class="fa fa-check"></i><b>7.1</b> Static Heatmap Layout 1</a></li>
<li class="chapter" data-level="7.2" data-path="heatmap-module.html"><a href="heatmap-module.html#static-heatmap-layout-2"><i class="fa fa-check"></i><b>7.2</b> Static Heatmap Layout 2</a></li>
<li class="chapter" data-level="7.3" data-path="heatmap-module.html"><a href="heatmap-module.html#interactive-heatmap"><i class="fa fa-check"></i><b>7.3</b> Interactive Heatmap</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="expression-plot-module.html"><a href="expression-plot-module.html"><i class="fa fa-check"></i><b>8</b> Expression Plot Module</a>
<ul>
<li class="chapter" data-level="8.1" data-path="expression-plot-module.html"><a href="expression-plot-module.html#set-up-options"><i class="fa fa-check"></i><b>8.1</b> Set up options</a></li>
<li class="chapter" data-level="8.2" data-path="expression-plot-module.html"><a href="expression-plot-module.html#data-table-1"><i class="fa fa-check"></i><b>8.2</b> Data Table</a></li>
<li class="chapter" data-level="8.3" data-path="expression-plot-module.html"><a href="expression-plot-module.html#result-table-1"><i class="fa fa-check"></i><b>8.3</b> Result Table</a></li>
<li class="chapter" data-level="8.4" data-path="expression-plot-module.html"><a href="expression-plot-module.html#rank-abundance-curve"><i class="fa fa-check"></i><b>8.4</b> Rank Abundance Curve</a></li>
<li class="chapter" data-level="8.5" data-path="expression-plot-module.html"><a href="expression-plot-module.html#laser-capture-microdissection"><i class="fa fa-check"></i><b>8.5</b> Laser Capture Microdissection</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="gene-set-enrichment-module.html"><a href="gene-set-enrichment-module.html"><i class="fa fa-check"></i><b>9</b> Gene Set Enrichment Module</a>
<ul>
<li class="chapter" data-level="9.1" data-path="gene-set-enrichment-module.html"><a href="gene-set-enrichment-module.html#gene-set-enrichment-analysis-gsea"><i class="fa fa-check"></i><b>9.1</b> Gene Set Enrichment Analysis (GSEA)</a></li>
<li class="chapter" data-level="9.2" data-path="gene-set-enrichment-module.html"><a href="gene-set-enrichment-module.html#over-representation-analysis-ora"><i class="fa fa-check"></i><b>9.2</b> Over-Representation Analysis (ORA)</a></li>
<li class="chapter" data-level="9.3" data-path="gene-set-enrichment-module.html"><a href="gene-set-enrichment-module.html#gene-expression"><i class="fa fa-check"></i><b>9.3</b> Gene Expression</a></li>
<li class="chapter" data-level="9.4" data-path="gene-set-enrichment-module.html"><a href="gene-set-enrichment-module.html#gene-set-heatmap"><i class="fa fa-check"></i><b>9.4</b> Gene Set Heatmap</a></li>
<li class="chapter" data-level="9.5" data-path="gene-set-enrichment-module.html"><a href="gene-set-enrichment-module.html#kegg-pathway-view"><i class="fa fa-check"></i><b>9.5</b> KEGG Pathway View</a></li>
<li class="chapter" data-level="9.6" data-path="gene-set-enrichment-module.html"><a href="gene-set-enrichment-module.html#wikipathway-view"><i class="fa fa-check"></i><b>9.6</b> WikiPathway View</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="pattern-clustering-module.html"><a href="pattern-clustering-module.html"><i class="fa fa-check"></i><b>10</b> Pattern Clustering Module</a>
<ul>
<li class="chapter" data-level="10.1" data-path="pattern-clustering-module.html"><a href="pattern-clustering-module.html#clustering-of-centroid-profiles"><i class="fa fa-check"></i><b>10.1</b> Clustering of Centroid Profiles</a></li>
<li class="chapter" data-level="10.2" data-path="pattern-clustering-module.html"><a href="pattern-clustering-module.html#optimal-number-of-clusters"><i class="fa fa-check"></i><b>10.2</b> Optimal Number of Clusters</a></li>
<li class="chapter" data-level="10.3" data-path="pattern-clustering-module.html"><a href="pattern-clustering-module.html#data-table-2"><i class="fa fa-check"></i><b>10.3</b> Data Table</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="wgcna-module.html"><a href="wgcna-module.html"><i class="fa fa-check"></i><b>11</b> WGCNA Module</a>
<ul>
<li class="chapter" data-level="11.1" data-path="wgcna-module.html"><a href="wgcna-module.html#dendrogram"><i class="fa fa-check"></i><b>11.1</b> Dendrogram</a></li>
<li class="chapter" data-level="11.2" data-path="wgcna-module.html"><a href="wgcna-module.html#gene-clusters"><i class="fa fa-check"></i><b>11.2</b> Gene Clusters</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="correlation-network-module.html"><a href="correlation-network-module.html"><i class="fa fa-check"></i><b>12</b> Correlation Network Module</a>
<ul>
<li class="chapter" data-level="12.1" data-path="correlation-network-module.html"><a href="correlation-network-module.html#visnetwork"><i class="fa fa-check"></i><b>12.1</b> visNetwork</a></li>
<li class="chapter" data-level="12.2" data-path="correlation-network-module.html"><a href="correlation-network-module.html#networkd3"><i class="fa fa-check"></i><b>12.2</b> networkD3</a></li>
<li class="chapter" data-level="12.3" data-path="correlation-network-module.html"><a href="correlation-network-module.html#selected-ids-data-table"><i class="fa fa-check"></i><b>12.3</b> Selected Ids Data Table</a></li>
<li class="chapter" data-level="12.4" data-path="correlation-network-module.html"><a href="correlation-network-module.html#correlation-summary"><i class="fa fa-check"></i><b>12.4</b> Correlation Summary</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="venn-diagram-module.html"><a href="venn-diagram-module.html"><i class="fa fa-check"></i><b>13</b> Venn Diagram Module</a>
<ul>
<li class="chapter" data-level="13.1" data-path="venn-diagram-module.html"><a href="venn-diagram-module.html#venn-diagram"><i class="fa fa-check"></i><b>13.1</b> Venn Diagram</a></li>
<li class="chapter" data-level="13.2" data-path="venn-diagram-module.html"><a href="venn-diagram-module.html#intersection-output"><i class="fa fa-check"></i><b>13.2</b> Intersection Output</a></li>
<li class="chapter" data-level="13.3" data-path="venn-diagram-module.html"><a href="venn-diagram-module.html#upsets"><i class="fa fa-check"></i><b>13.3</b> UpSets</a></li>
<li class="chapter" data-level="13.4" data-path="venn-diagram-module.html"><a href="venn-diagram-module.html#upsets-table"><i class="fa fa-check"></i><b>13.4</b> UpSets Table</a></li>
</ul></li>
<li class="chapter" data-level="14" data-path="pcsf-module.html"><a href="pcsf-module.html"><i class="fa fa-check"></i><b>14</b> PCSF Module</a>
<ul>
<li class="chapter" data-level="14.1" data-path="pcsf-module.html"><a href="pcsf-module.html#pcsf-network"><i class="fa fa-check"></i><b>14.1</b> PCSF network</a></li>
<li class="chapter" data-level="14.2" data-path="pcsf-module.html"><a href="pcsf-module.html#pcsf-function-cluster"><i class="fa fa-check"></i><b>14.2</b> PCSF Function Cluster</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="dose-response-module.html"><a href="dose-response-module.html"><i class="fa fa-check"></i><b>15</b> Dose Response Module</a>
<ul>
<li class="chapter" data-level="15.1" data-path="dose-response-module.html"><a href="dose-response-module.html#fitting-curve"><i class="fa fa-check"></i><b>15.1</b> Fitting Curve</a>
<ul>
<li class="chapter" data-level="15.1.1" data-path="dose-response-module.html"><a href="dose-response-module.html#fitting-curve-1"><i class="fa fa-check"></i><b>15.1.1</b> Fitting Curve</a></li>
<li class="chapter" data-level="15.1.2" data-path="dose-response-module.html"><a href="dose-response-module.html#result-table-and-data-table"><i class="fa fa-check"></i><b>15.1.2</b> Result Table and Data Table</a></li>
<li class="chapter" data-level="15.1.3" data-path="dose-response-module.html"><a href="dose-response-module.html#model-selection"><i class="fa fa-check"></i><b>15.1.3</b> Model Selection</a></li>
</ul></li>
<li class="chapter" data-level="15.2" data-path="dose-response-module.html"><a href="dose-response-module.html#result-table-all"><i class="fa fa-check"></i><b>15.2</b> Result Table (all)</a></li>
<li class="chapter" data-level="15.3" data-path="dose-response-module.html"><a href="dose-response-module.html#browsing"><i class="fa fa-check"></i><b>15.3</b> Browsing</a></li>
</ul></li>
<li class="chapter" data-level="16" data-path="time-course-module.html"><a href="time-course-module.html"><i class="fa fa-check"></i><b>16</b> Time Course Module</a>
<ul>
<li class="chapter" data-level="16.1" data-path="time-course-module.html"><a href="time-course-module.html#fitting-curve-2"><i class="fa fa-check"></i><b>16.1</b> Fitting Curve</a>
<ul>
<li class="chapter" data-level="16.1.1" data-path="time-course-module.html"><a href="time-course-module.html#fitting-curve-3"><i class="fa fa-check"></i><b>16.1.1</b> Fitting Curve</a></li>
<li class="chapter" data-level="16.1.2" data-path="time-course-module.html"><a href="time-course-module.html#result-table-and-data-table-1"><i class="fa fa-check"></i><b>16.1.2</b> Result Table and Data Table</a></li>
</ul></li>
<li class="chapter" data-level="16.2" data-path="time-course-module.html"><a href="time-course-module.html#result-table-all-1"><i class="fa fa-check"></i><b>16.2</b> Result Table (all)</a></li>
<li class="chapter" data-level="16.3" data-path="time-course-module.html"><a href="time-course-module.html#browsing-1"><i class="fa fa-check"></i><b>16.3</b> Browsing</a></li>
</ul></li>
<li class="chapter" data-level="17" data-path="monotonic-trend-module.html"><a href="monotonic-trend-module.html"><i class="fa fa-check"></i><b>17</b> Monotonic Trend Module</a>
<ul>
<li class="chapter" data-level="17.1" data-path="monotonic-trend-module.html"><a href="monotonic-trend-module.html#fitting-curve-4"><i class="fa fa-check"></i><b>17.1</b> Fitting Curve</a>
<ul>
<li class="chapter" data-level="17.1.1" data-path="monotonic-trend-module.html"><a href="monotonic-trend-module.html#fitting-curve-5"><i class="fa fa-check"></i><b>17.1.1</b> Fitting Curve</a></li>
<li class="chapter" data-level="17.1.2" data-path="monotonic-trend-module.html"><a href="monotonic-trend-module.html#result-table-and-data-table-2"><i class="fa fa-check"></i><b>17.1.2</b> Result Table and Data Table</a></li>
</ul></li>
<li class="chapter" data-level="17.2" data-path="monotonic-trend-module.html"><a href="monotonic-trend-module.html#result-table-all-2"><i class="fa fa-check"></i><b>17.2</b> Result Table (all)</a></li>
<li class="chapter" data-level="17.3" data-path="monotonic-trend-module.html"><a href="monotonic-trend-module.html#browsing-2"><i class="fa fa-check"></i><b>17.3</b> Browsing</a></li>
</ul></li>
<li class="chapter" data-level="18" data-path="merge-data-module.html"><a href="merge-data-module.html"><i class="fa fa-check"></i><b>18</b> Merge Data Module</a>
<ul>
<li class="chapter" data-level="18.1" data-path="merge-data-module.html"><a href="merge-data-module.html#merge-data"><i class="fa fa-check"></i><b>18.1</b> Merge Data</a></li>
<li class="chapter" data-level="18.2" data-path="merge-data-module.html"><a href="merge-data-module.html#data-source"><i class="fa fa-check"></i><b>18.2</b> Data Source</a></li>
</ul></li>
<li class="chapter" data-level="19" data-path="output-module.html"><a href="output-module.html"><i class="fa fa-check"></i><b>19</b> Output Module</a></li>
<li class="chapter" data-level="20" data-path="introduction-to-multiple-testing-adjustment-in-genomic-data-analysis.html"><a href="introduction-to-multiple-testing-adjustment-in-genomic-data-analysis.html"><i class="fa fa-check"></i><b>20</b> Introduction to Multiple Testing Adjustment in Genomic Data Analysis</a>
<ul>
<li class="chapter" data-level="20.1" data-path="introduction-to-multiple-testing-adjustment-in-genomic-data-analysis.html"><a href="introduction-to-multiple-testing-adjustment-in-genomic-data-analysis.html#differentially-expressed-gene-deg-analysis"><i class="fa fa-check"></i><b>20.1</b> Differentially Expressed Gene (DEG) Analysis</a></li>
<li class="chapter" data-level="20.2" data-path="introduction-to-multiple-testing-adjustment-in-genomic-data-analysis.html"><a href="introduction-to-multiple-testing-adjustment-in-genomic-data-analysis.html#over-representation-analysis-ora-1"><i class="fa fa-check"></i><b>20.2</b> Over Representation Analysis (ORA)</a></li>
<li class="chapter" data-level="20.3" data-path="introduction-to-multiple-testing-adjustment-in-genomic-data-analysis.html"><a href="introduction-to-multiple-testing-adjustment-in-genomic-data-analysis.html#gene-set-enrichment-analysis-gsea-1"><i class="fa fa-check"></i><b>20.3</b> Gene Set Enrichment Analysis (GSEA)</a></li>
<li class="chapter" data-level="20.4" data-path="introduction-to-multiple-testing-adjustment-in-genomic-data-analysis.html"><a href="introduction-to-multiple-testing-adjustment-in-genomic-data-analysis.html#correlation-network-analysis"><i class="fa fa-check"></i><b>20.4</b> Correlation Network Analysis</a></li>
</ul></li>
<li class="chapter" data-level="21" data-path="appendix.html"><a href="appendix.html"><i class="fa fa-check"></i><b>21</b> Appendix</a>
<ul>
<li class="chapter" data-level="21.1" data-path="appendix.html"><a href="appendix.html#scalability-testing"><i class="fa fa-check"></i><b>21.1</b> Scalability testing</a></li>
<li class="chapter" data-level="21.2" data-path="appendix.html"><a href="appendix.html#feature-comparison"><i class="fa fa-check"></i><b>21.2</b> Feature comparison</a></li>
</ul></li>
<li class="chapter" data-level="22" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i><b>22</b> References</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">xOmicsShiny: an R shiny application for cross-omics data analysis and pathway mapping</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="wgcna-module" class="section level1 hasAnchor" number="11">
<h1><span class="header-section-number">Chapter 11</span> WGCNA Module<a href="wgcna-module.html#wgcna-module" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>This module is implemented by the WGCNA R package, which is a tool to build weighted gene correlation networks for expression data. The module contains two functional tabs: Dendrogram and Gene Clusters.</p>
<div id="dendrogram" class="section level2 hasAnchor" number="11.1">
<h2><span class="header-section-number">11.1</span> Dendrogram<a href="wgcna-module.html#dendrogram" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>If the data set is a the saved projects in the CSV file, the precomputed results will be directly loaded based on up to 10,000 genes with default parameter values. For other data set or if the user want to re-run WGCNA on the saved project with a different parameter or number of genes, just click ‘Re-run’ and wait for a few minutes, the results will show up once the analysis is done.</p>
<p><a href="Figures/f11-1.png"><img src="Figures/f11-1.png" /></a></p>
</div>
<div id="gene-clusters" class="section level2 hasAnchor" number="11.2">
<h2><span class="header-section-number">11.2</span> Gene Clusters<a href="wgcna-module.html#gene-clusters" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>This tab shows a searchable table that listing all the genes for each cluster. User can use the copy button to get the complete gene list of cluster and check with other tools, for example, Heatmap.</p>
<p><a href="Figures/f11-2.png"><img src="Figures/f11-2.png" /></a></p>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="pattern-clustering-module.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="correlation-network-module.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/interactivereport/xOmicsShiny/edit/gh-pages/11-WGCNAModule.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": "https://github.com/interactivereport/xOmicsShiny/blob/gh-pages/11-WGCNAModule.Rmd",
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
